openssl-sys/src/libressl.rs

```diff
@@ -525,6 +579,7 @@ extern {
                                                                 is_export: c_int,
                                                                 keylength: c_int)
                                                                 -> *mut ::EC_KEY);
+    pub fn SSL_get_session(ssl: *mut ::SSL) -> *mut ::SSL_SESSION;
```

Comment:
Yes! I missed that. Thanks.
There's another issue with the size of SSL_SESSION and crypto_ex_data_st. Four bits off from their C representations.